Dockerfile

```diff
@@ -0,0 +1,22 @@
+FROM python:3.9-slim-buster as base
+# Install poetry
+RUN pip install poetry
+WORKDIR /app
+# Expose the port that we want the host to listen to
+EXPOSE 5000
+# Copy across the project files
+COPY . /app
+# Install the project dependencies
+RUN poetry install
+
+FROM base as production
+ENV FLASK_ENV=production
+# Set the script to run as production on startup of the container
+ENTRYPOINT ["poetry", "run", "gunicorn", "--config", "gunicorn_config.py", "todo_app.app:create_app()"]
+
+FROM base as development
+ENV FLASK_ENV=development
+# Expose the port that we want the host to listen to
+EXPOSE 5001
+# Set the script to run as development on startup of the container
+ENTRYPOINT ["poetry", "run", "flask", "run", "--host=0.0.0.0", "--port=5001"]
```

Comment:
Note that there's no _need_ that this has to be on port 5001; the dev container will be run separately to the prod one, so we could leave both using their internal 5000, and we just map our host's ports separately, e.g. 5000:5000 for prod and 5001:5000 for dev. There's nothing wrong with this approach at all, just making that clear in case it isn't!